File: leetcode/Easy/Two Sum/two_sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """

        for i in range(len(nums)):
            for j in range(i+1,len(nums)):
                if (nums[i] + nums[j]) == target:
                    return [i,j]

        # from itertools import combinations

        # for i, j in combinations(range(len(nums)), 2):
        #     if nums[i] + nums[j] == target:
        #         return [i, j]

File: leetcode/Easy/Two Sum/Two Sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
  const seen = {};
  for (let i = 0; i < nums.length; i++) {
    const y = target - nums[i];

    if (y in seen) {
      return [seen[y], i];
    }

    seen[nums[i]] = i;
  }
  return [];
};
